src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToControlGeneratedAssemblyInfo.cs

```diff
@@ -595,5 +595,52 @@ public void GenerateUserSecretsForTestProject()
 
             AssemblyInfo.Get(assemblyPath)["UserSecretsIdAttribute"].Should().Be("SecretsIdValue");
         }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void It_includes_repository_url(bool privateRepo)
+        {
+            var fakeUrl = "fakeUrl";
+            var testAsset = _testAssetsManager
+                .CopyTestAsset("HelloWorld")
```

Comment:
If you use `TestProject` to create your test project, you can just use the `AdditionalProperties` property to set the repository URL MSBuild properties instead of having to call `WithProjectChanges` and create XML elements.